apps/core/src/hooks/useGetClockTimestamp.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useIotaClient } from '@iota/dapp-kit';
+import { IotaClient } from '@iota/iota-sdk/client';
+import { useQuery } from '@tanstack/react-query';
+import { CLOCK_PACKAGE_ID } from '../constants/clock.constants';
+
+type ClockFields = {
+    id: {
+        id: string;
+    };
+    timestamp_ms: string;
+};
+
+export function useGetClockTimestamp() {
+    const client = useIotaClient();
+    const { data } = useQuery({
+        queryKey: ['get-clock-timestamp', client],
+        queryFn: async () => getClockTimestamp(client),
+        staleTime: 10 * 1000,
+        refetchInterval: 10 * 1000, // refetch every 10 seconds to keep the clock updated but not overload the servery,
+        placeholderData: Infinity,
+    });
+
+    return {
+        data: data!,
+    };
```

Comment:
we could now simply return the query
```suggestion
    return useQuery({
        queryKey: ['get-clock-timestamp', client],
        queryFn: async () => getClockTimestamp(client),
        staleTime: 10 * 1000,
        refetchInterval: 10 * 1000, // refetch every 10 seconds to keep the clock updated but not overload the servery,
        placeholderData: Infinity,
    });
```

Reply:
then we'll have typescript issue because of data will be number or undefined.

Reply:
Under what situation could it be undefined? I wouldn't silence the undefined

Reply:
It doens't undefined as fact (because of we have placeholderData), but typescript think it might be undefined and highlight error.